tgui/packages/tgui/interfaces/Techweb.js

```diff
@@ -0,0 +1,820 @@
+  const { id, can_unlock, tier } = node;
+  const {
+    name,
+    description,
+    costs,
+    design_ids,
+    prereq_ids,
+  } = node_cache[id];
+  const [
+    techwebRoute,
+    setTechwebRoute,
+  ] = useLocalState(context, 'techwebRoute', null);
+  const [
+    tabIndex,
+    setTabIndex,
+  ] = useLocalState(context, 'nodeDetailTabIndex', 0);
+
+  const techcompl = prereq_ids
+    .filter(x => nodes.find(y => y.id === x)?.tier === 0)
+    .length;
+
+  return (
+    <Section
+      className="Techweb__NodeContainer"
+      title={name}
+      width={25}
+      fitted>
+      {!nocontrols && (
+        <>
+          {!nodetails && (
+            <Button
+              icon="tasks"
+              onClick={() => {
+                setTechwebRoute({ route: "details", selectedNode: id });
+                setTabIndex(0);
+              }}>
+              Details
+            </Button>
+          )}
+          {((tier > 0) && (!destructive)) && (!!researchable) && (
+            <Button
+              icon="lightbulb"
+              disabled={!can_unlock || tier > 1}
+              onClick={() => act("researchNode", { node_id: id })}>
+              Research
+            </Button>
+          )}
+          {destructive && (
+            <Button
+              icon="trash"
+              color="red"
+              onClick={() => act("destroyfortech", { node_id: id })}>
+              Destroy item for node
+            </Button>
+          )}
+        </>)}
+      {tier !== 0 && !!compact && !destructive && (
+        <Flex className="Techweb__NodeProgress">
+          {costs.map(k => {
+            const reqPts = Math.max(0, k.value);
+            const nodeProg = Math.min(reqPts, points[k.type]) || 0;
+            return (
+              <Flex.Item key={k.type} grow={1} basis={0}>
+                <ProgressBar
+                  ranges={{
+                    good: [0.5, Infinity],
+                    average: [0.25, 0.5],
+                    bad: [-Infinity, 0.25],
+                  }}
+                  value={reqPts === 0
+                    ? 1
+                    : Math.min(1, (points[k.type]||0) / reqPts)}>
+                  {abbreviateName(k.type)} ({nodeProg}/{reqPts})
+                </ProgressBar>
+              </Flex.Item>
+            );
+          })}
+        </Flex>
+      )}
+      <Box className="Techweb__NodeDescription">
+        {description}
+      </Box>
+      {!!compact && (
+        <Box className="Techweb__NodeUnlockedDesigns" mt={1}>
+          {design_ids.map((k, i) => (
+            <Box
+              key={id}
+              className={`${design_cache[k].class} Techweb__DesignIcon`}
+              // Uncomment these only when tooltip performance is improved.
+              // Reason for removal: Causes massive performance decreases
+              // tooltip={design_cache[k].name}
+              // tooltipPosition={i % 15 < 7 ? "right" : "left"}
```

Comment:
I think the tooltips can now be enabled since https://github.com/BeeStation/BeeStation-Hornet/pull/5415 has been merged, though should test performance just in case.